src/Microsoft.IdentityModel.Tokens/Exceptions/SecurityTokenInvalidSigningKeyException.cs

```diff
@@ -73,6 +73,8 @@ public SecurityTokenInvalidSigningKeyException(string message, Exception inner)
         protected SecurityTokenInvalidSigningKeyException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
+            //TODO - investigate if SigningKey type should be serialized
```

Comment:
For the serializer to work with the SigningKey, the abstract base class (SecurityKey) would also need to be marked as serializable as well as the types on SecurityKey. The list of types that would need to be marked as serializable would be:

- CryptoProviderCache
- CryptoProviderFactory
- InMemoryCryptoProviderCache
- SecurityKey